src/Environment.hs

```diff
@@ -0,0 +1,52 @@
+module Environment where
+
+import           Syntax
+import           ValueSyntax
+
+import qualified Data.Map                      as M
+import qualified Data.Text                     as T
+
+-- TODO: Think about Environment structure. How modules should be stored etc.
+
+emptyEnvironment :: Environment
+emptyEnvironment = Environment M.empty
+
+searchEnvironment :: T.Text -> Environment -> Value
+searchEnvironment name (Environment env) = env M.! name
+
+lookupIdentifier :: Identifier -> Environment -> Value
+lookupIdentifier (Identifier name) = searchEnvironment name
+
+lookupTypeVar :: TypeVar -> Environment -> Value
+lookupTypeVar (TypeVar name) = searchEnvironment name
+
+lookupConstructorName :: ConstructorName -> Environment -> Value
+lookupConstructorName (ConstructorName name) = searchEnvironment name
+
+lookupTypeName :: TypeName -> Environment -> Value
+lookupTypeName (TypeName name) = searchEnvironment name
+
+-- lookupModName :: ModName -> Environment -> Value
+-- lookupModName (ModName name) = searchEnvironment name
+
+lookupQualifiedVar :: ModName -> Identifier -> Environment -> Value
+lookupQualifiedVar modName identifier env = undefined
+
+lookupQualifiedConstructor :: ModName -> ConstructorName -> Environment -> Value
+lookupQualifiedConstructor modName constructorName env = undefined 
+
+lookupInternal :: Identifier -> Environment -> Value
+lookupInternal identifier env = undefined
```

Comment:
The internal functions should be Haskell functions which take lists of arguments as the input.
Lets take `addInt` internal value as an example:
```haskell
addInt :: [Value] -> IO Value
addInt [a, b] = return a + b
addInt _ = error "Invalid argument for 'addInt'"
```